sdk/core/AzureCore/Tests/UtilTests.swift

```diff
@@ -69,7 +69,7 @@ class RegexUtilTests: XCTestCase {
     }
 
     func testStringUtil() {
-        let dateString = "Jan, 02 2020 07:12:34 GMT"
+        let dateString = "Thu, 02 Jan 2020 07:12:34 GMT"
```

Comment:
This format was carried over from (presumably) the Python SDK. If what it called RFC1123 isn't actually RFC1123, that's fine, but if we have changed the format associated with these methods, they will no longer work with the Storage SDK formats that code was written for. I suspect it would return nil.

Reply:
I noticed the tests were failing because that is not actually an RFC1123 string. The change shows a valid RFC1123 string. The original format is very odd because why would there be a comma after the short month name? So is Storage just using a very stupid custom date format based on someone misunderstanding RFC1123?

Going to research what the other SDKs do. Also if this ***is*** a custom format for storage, we should move these methods out of Core and into the Storage framework.

Reply:
My change also matches the format specified in `StorageBlobClient.StorageJSONEncoder`.

Reply:
JS seems like it expects the standard RFC1123 format: https://github.com/Azure/azure-sdk-for-js/blob/96de036571c46659938ace951a8eb78834b8dbc4/sdk/core/core-http/test/serializationTests.ts#L283

Reply:
Ah, okay, then the problem must have been me! Thanks for checking!

Reply:
I changed around how we implemented the various conversions that live in the Utils folder in 8be84e4. The extension properties felt weirdly non-swifty, so I replaced them with converting initializers. It feels a lot better to me now but very interested to hear your feedback.